core-bundle/src/Resources/contao/library/Contao/Idna.php

```diff
@@ -101,14 +101,29 @@ public static function encodeEmail($strEmail)
 		}
 
 		$arrChunks = explode('@', $strEmail);
-		$strHost = static::encode(array_pop($arrChunks));
+		$strHost = array_pop($arrChunks);
 
 		if ($strHost == '')
 		{
 			return '';
 		}
 
-		return implode('@', $arrChunks) . '@' . $strHost;
+		$strQuery = null;
+
+		// Strip the query string (see #2149)
+		if (strpos($strHost, '?') !== false)
+		{
+			list($strHost, $strQuery) = explode('?', $strHost);
```

Comment:
```suggestion
			list($strHost, $strQuery) = explode('?', $strHost, 2);
```

This way we would not strip occurrences of `?`s in the query itself.